### ISSUE TITLE
Unable to create Generic overloads when Extends type with Module Name

### ISSUE THREAD

**Issue by daryllabar** [Microsoft/TypeScript#4649]
I'm running VS 2013.  In the following example, the NoWorks Module has errors on the first to overload signatures.  The Works doesn't have any issues:

```typescript
module Common {
    export interface A { }
    export interface B { }
}

module NoWorks {
    class Example {

        static helpMe<T extends Common.A>(value: string): string;
        static helpMe<T extends Common.B>(value: string): number;
        static helpMe(value: string): any {
            return null;
        }
    }
}

module Works {
    import A = Common.A;
    import B = Common.B;

    class Example {

        static helpMe<T extends A>(value: string): string;
        static helpMe<T extends B>(value: string): number;
        static helpMe(value: string): any {
            return null;
        }
    }
}
```

**Comment by vladima** [comment]
What version of TypeScript do you use? [Version 1.6-beta](http://download.microsoft.com/download/4/4/3/443F86B7-A89F-48E6-AC96-0AAC2A910A29/TS1.6-Beta-VUOOB.40831.00/TypeScript_Dev12.exe) does not show any errors for me?

**Comment by daryllabar** [comment]
That's probably my issue.  I'm on 1.5.  I just started using Typescript this week and couldn't figure out what the heck was happening.

**Comment by mhegazy** [comment]
@daryllabar is this still an issue? or did 1.6 solve the problem?

**Comment by daryllabar** [comment]
@mhegazy I don't know.  I'm not using 1.6  according to @vladima it works in 1.6-beta.

**Comment by mhegazy** [comment]
I tried this on 1.5.3, and i am not seeing issues.. i think there is something more to this.. see [this code in 1.5 playground](http://www.typescriptlang.org/Playground#src=module%20Common%20%7B%0D%0A%20%20%20%20export%20interface%20A%20%7B%20%7D%0D%0A%20%20%20%20export%20interface%20B%20%7B%20%7D%0D%0A%7D%0D%0A%0D%0Amodule%20NoWorks%20%7B%0D%0A%20%20%20%20class%20Example%20%7B%0D%0A%20%20%20%20%20%20%20%20static%20helpMe%3CT%20extends%20Common.A%3E(value%3A%20string)%3A%20string%3B%0D%0A%20%20%20%20%20%20%20%20static%20helpMe%3CT%20extends%20Common.B%3E(value%3A%20string)%3A%20number%3B%0D%0A%20%20%20%20%20%20%20%20static%20helpMe(value%3A%20string)%3A%20any%20%7B%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20return%20null%3B%0D%0A%20%20%20%20%20%20%20%20%7D%0D%0A%20%20%20%20%7D%0D%0A%7D%0D%0A%0D%0Amodule%20Works%20%7B%0D%0A%20%20%20%20import%20A%20%3D%20Common.A%3B%0D%0A%20%20%20%20import%20B%20%3D%20Common.B%3B%0D%0A%0D%0A%20%20%20%20class%20Example%20%7B%0D%0A%0D%0A%20%20%20%20%20%20%20%20static%20helpMe%3CT%20extends%20A%3E(value%3A%20string)%3A%20string%3B%0D%0A%20%20%20%20%20%20%20%20static%20helpMe%3CT%20extends%20B%3E(value%3A%20string)%3A%20number%3B%0D%0A%20%20%20%20%20%20%20%20static%20helpMe(value%3A%20string)%3A%20any%20%7B%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20return%20null%3B%0D%0A%20%20%20%20%20%20%20%20%7D%0D%0A%20%20%20%20%7D%0D%0A%7D)

**Comment by daryllabar** [comment]
Maybe it is an issue with the VS support?

I just upgraded to 2015 yesterday, and I'm getting the same error:

![screenshot](https://cloud.githubusercontent.com/assets/9728423/9744109/e906f4e2-5637-11e5-8d43-ff69e9bb7fef.jpg)

**Comment by mhegazy** [comment]
thanks @daryllabar for getting to the bottom of this issues.

**Comment by daryllabar** [comment]
@mhegazy I'm just Sorry I incorrectly submitted the issue.  First week on Typescript and didn't know what was causing what to happen.

**Comment by mhegazy** [comment]
no worries. glad the issue is resolved. feel free to ping either here or on the [gitter typescript chat room](https://gitter.im/Microsoft/TypeScript) if you have any questions.
